### PR TITLE
use digestif 1.2.0 API

### PR DESCRIPTION
**Pull request by hannesm**

### CODE REVIEW

**Comment by hannesm** [comment]
//cc @reynir and @dinosaure

File: pk/rsa.ml [hunk 1]
```diff
@@ -274,69 +269,26 @@ module PKCS1 = struct
...
-         Some (`SHA256, sha256)
-       else if is_prefix sha384 buf then
-         Some (`SHA384, sha384)
-       else if is_prefix sha512 buf then
-         Some (`SHA512, sha512)
-       else
-         None)
+    (fun h ->
+       match List.assoc_opt h map with
+       | None -> invalid_arg "unsupported hash (only MD5 and SHA are supported)"
```
Comment (reynir) [inline pk/rsa.ml:284]: Considering digestif implements SHA3 too...
```suggestion
       | None -> invalid_arg "unsupported hash (only MD5, SHA1 and SHA2 are supported)"
```

**Review by reynir — APPROVED** [review]

File: pk/rsa.ml (hunk 1 above)
Comment (hannesm) [reply pk/rsa.ml]: Thanks for the suggestion. This morning while riding my bike I figured that we should avoid these exceptions, and provide a more narrow interface. So I went ahead and pushed c19f54e -- let me know what you think of that. :)

File: pk/mirage_crypto_pk.mli [hunk 2]
```diff
@@ -187,15 +187,12 @@ module Rsa : sig
         was produced with the given [key] as per {{!sig_encode}sig_encode}, or
         [None] *)
 
-    type hash = [ `MD5 | `SHA1 | `SHA224 | `SHA256 | `SHA384 | `SHA512 ]
-    (** The type of supported hash algorithms. *)
-
-    val min_key : hash -> bits
+    val min_key : [ `MD5 | `SHA1 | `SHA224 | `SHA256 | `SHA384 | `SHA512 ] -> bits
```
Comment (reynir) [inline pk/mirage_crypto_pk.mli:190]: This is great! We can be more clear that these are a subset of `Digestif.hash'`:
```suggestion
    val min_key : [< Digestif.hash' > `MD5 `SHA1 `SHA224 `SHA256 `SHA384 `SHA512 ] -> bits
```

**Review by reynir — APPROVED** [review]

File: pk/mirage_crypto_pk.mli (hunk 2 above)
Comment (hannesm) [reply pk/mirage_crypto_pk.mli]: oh yes, that's beatiful. I applied it here and on the other locations. 367366b

**Pull request merged**